builder/dockerfile/builder.go

```diff
@@ -119,6 +121,12 @@ func NewBuilder(clientCtx context.Context, config *types.ImageBuildOptions, back
 		config.BuildArgs = make(map[string]string)
 	}
 	ctx, cancel := context.WithCancel(clientCtx)
+
+	// If no backend is provided, there will be no caching
+	var imageCacheForBuild builder.ImageCacheForBuild
+	if backend != nil {
+		imageCacheForBuild = backend.(builder.ImageCache).MakeImageCacheForBuild(config.CacheFrom)
```

Comment:
Check the interface instead if you want to make this optional. Previous code also has this optional with a `ok` check.

`ImageCacheForBuild->ImageCache`
`ImageCache->ImageCacheMaker`

Until we don't do registry check I don't think we actually need state at all.